plugins/modules/lvol.py

```diff
@@ -236,12 +262,118 @@ def create_lv(module, name):
+    # Csize: <number of current LPs in LV>
+    # PPsize: <PP size of each LP>
+    cmd = "lquerylv -L %s -cst" % lvid
+    fail_msg = "Failed to fetch the physical partition size and current \
+        number of logical partitions in the logical volume"
+    lv_run_cmd(module, cmd, None, fail_msg, fetch=True)
+    pattern = r"^Csize:\s+(\d+)"
+    curr_lps_in_lv = int(re.search(pattern, result['stdout'], re.MULTILINE).group(1))
+    pattern = r"^PPsize:\s+(\d+)"
+    pp_size = int(re.search(pattern, result['stdout'], re.MULTILINE).group(1))
+
+    # check if plus (+) sign is used with the 'size' parameter
+    size = module.params['size']
+    plus_sign_used = False
+    if size[0] == "+":
+        plus_sign_used = True
+        size = size[1:]  # remove plus sign
+    elif size[0] == "-":
+        # minus sign is not supported
+        result['msg'] += "\nMinus sign (-) is not a supported prefix for 'size' parameter."
+        module.fail_json(**result)
+
+    # convert 'size' parameter to corresponding number of LPs
+    # needed to satisfy the specified new size if suffixes are
+    # used (B, b, K, k, M, m, G, g)
+    # calculation logic lifted verbatim from extendlv cmd in lvm
+    pp_size = 1 << pp_size
+    if (size[-1] == "G") or (size[-1] == "g"):
+        shift = 30
+    elif (size[-1] == "M") or (size[-1] == "m"):
+        shift = 20
+    elif (size[-1] == "K") or (size[-1] == "k"):
+        shift = 10
+    elif (size[-1] == "B") or (size[-1] == "b"):
+        shift = 9
+    else:  # no suffix
+        shift = 0
+
+    # conversion from size with suffix to correspoding num
+    # of LPs to satisfy the size
+    if shift != 0:
+        size = int(size[:-1])
+        size = (size * (1 << shift))
+        size = (size + pp_size - 1) / pp_size
+    size = int(size)
+
+    # calculate how much to extend the LV on each scenario
+    # (1) if the calculated new size is smaller than the
+    # current size, then fail (without + sign)
+    # (2) if the plus sign is USED, then add 'size' LPs
+    # to the existing number of LPs in the LV
+    # (3) if the plus sign is NOT used then extend current
+    # LV until it satisfies the new specified number of
+    # 'size' LPs
+    # (4) if plus sign is NOT used AND expected new 'size'
+    # is equal to current size, then no changes needed.
+    fail_msg = "\nFailed to extend logical volume '%s'." % name
+    if not plus_sign_used and size < curr_lps_in_lv:
+        result['cmd'] = ""
+        result['msg'] += "\nReducing the size of the logical volume is not supported."
+        module.fail_json(**result)
+    elif plus_sign_used:
+        cmd = "extendlv %s %s" % (name, size)
+        success_msg = "\nLogical volume '%s' has been extended by %s." % \
+            (name, module.params['size'])
+        lv_run_cmd(module, cmd, success_msg, fail_msg, None)
+    elif size != curr_lps_in_lv:
+        # calculate how much to extend in order to satisfy 'size'
+        size = size - curr_lps_in_lv
```

Comment:
same here, should extendlv command self already handle this ?

Reply:
extendlv does not calculate the difference between current and specified size.
extendlv will only ADD the specified size on top of what size already exist in the lv